File: Oxiyopal.Models/Factura.cs
using Oxiyopal.Models.Base;
using Oxiyopal.Models.ViewModels;
using System;
using System.Collections.Generic;
using System.Text;

namespace Oxiyopal.Models
{
    public class Factura: EntityBase<int>
    {
        public int Valor { get; set; }
        public int Serial { get; set; }
        public DateTime FechaDeVenta { get; set; }
        public int ClienteId { get; set; }
        public Cliente Cliente { get; set; }
        public IList<HistorialDeFactura> HistorialDeFacturas { get; set; }
    }
}

File: Oxiyopal.WebUI/ClientApp/src/app/models/Factura.ts
import { EntityBase } from './Base/EntityBase';
import { HistorialDeFactura } from './ViewModels/HistorialDeFactura';
import { Cliente } from './Cliente';

export interface Factura extends EntityBase<number> {
    valor: number;
    serial: number;
    fechaDeVenta: Date | string;
    clienteId: number;
    cliente: Cliente;
    historialDeFacturas: HistorialDeFactura[];
}

